exam001/Views/RequestDetailsViewController.swift

```diff
@@ -0,0 +1,334 @@
+    }
+    
+    private func setupBasketImageConstraints() {
+        basketImageView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            basketImageView.topAnchor.constraint(equalTo: locationLabel.bottomAnchor, constant: Config.imageTopAnchor),
+            basketImageView.centerXAnchor.constraint(equalTo: containerView.centerXAnchor)
+            ])
+    }
+    
+    private func setupItemsLabelConstraints() {
+        itemsLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            itemsLabel.topAnchor.constraint(equalTo: basketImageView.bottomAnchor, constant: Config.labelTopAnchor),
+            itemsLabel.leftAnchor.constraint(equalTo: containerView.leftAnchor, constant: Config.subviewsHorizontalAnchor),
+            itemsLabel.rightAnchor.constraint(equalTo: containerView.rightAnchor, constant: -Config.subviewsHorizontalAnchor)
+            ])
+    }
+    
+    private func setupNameAndFloorImageConstraints() {
+        nameAndFloorImageView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            nameAndFloorImageView.topAnchor.constraint(equalTo: itemsLabel.bottomAnchor, constant: Config.imageTopAnchor),
+            nameAndFloorImageView.centerXAnchor.constraint(equalTo: containerView.centerXAnchor)
+            ])
+    }
+    
+    private func setupNameAndFloorLabelConstraints() {
+        nameAndFloorLabel.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            nameAndFloorLabel.topAnchor.constraint(equalTo: nameAndFloorImageView.bottomAnchor, constant: Config.labelTopAnchor),
+            nameAndFloorLabel.leftAnchor.constraint(equalTo: containerView.leftAnchor, constant: Config.subviewsHorizontalAnchor),
+            nameAndFloorLabel.rightAnchor.constraint(equalTo: containerView.rightAnchor, constant: -Config.subviewsHorizontalAnchor)
+            ])
+    }
+    
+    private func setupFillerViewConstraints() {
+        fillerView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            fillerView.rightAnchor.constraint(equalTo: containerView.rightAnchor),
+            fillerView.leftAnchor.constraint(equalTo: containerView.leftAnchor),
+            fillerView.topAnchor.constraint(equalTo: nameAndFloorLabel.bottomAnchor, constant: Config.fillerViewTopAnchor),
+            fillerView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+            ])
+    }
+    
+}
+
+extension RequestDetailsViewController: DetailModelDelegate {
+    
+    func checkStatus() {
+        requestStatus = "Taken"
+        if detailVM!.requestStatus != "Available" {
+            button.isEnabled = false
+            button.alpha = 0.5
+            button.titleLabel?.text = "Taken"
+        } else {
+            button.isEnabled = true
+            button.alpha = 1.0
+            button.titleLabel?.text = "Take"
```

Comment:
Sometimes we can use linear conditional statements
Example:
`button.isEnabled` can be assigned  as below
`button.isEnabled = detailVM.requestStatus != "Available" ? false : true`